src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs

```diff
@@ -69,100 +68,61 @@ internal partial class SQLitePersistentStorage
 
         private async Task FlushSpecificWritesAsync<TKey>(
             SqlConnection connection, MultiDictionary<TKey, Action<SqlConnection>> keyToWriteActions,
-            Dictionary<TKey, CountdownEvent> keyToCountdown, TKey key, 
+            Dictionary<TKey, Task> keyToWriteTask, TKey key,
             ArrayBuilder<Action<SqlConnection>> writesToProcess,
```

Comment:
the diff below is fairly awful.  i recommend just looking at the final form as the logic is basically completely rewritten form the initial form. 

The initial form was so complex because flushing was previously synchronous.  So if one thread was flushing, newly arriving threads needed to know that another thread was flushing, and needing to wait on that finishing.

Now that flushing is async, it's much simpler to mark that writes are happening, and much easier for newly arriving threads to 'await' that work.